// @polar-sh/sdk/dist/commonjs/models/components/organizationcustomeremailsettings.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.OrganizationCustomerEmailSettings$outboundSchema = exports.OrganizationCustomerEmailSettings$inboundSchema = void 0;
exports.organizationCustomerEmailSettingsToJSON = organizationCustomerEmailSettingsToJSON;
exports.organizationCustomerEmailSettingsFromJSON = organizationCustomerEmailSettingsFromJSON;
const z = __importStar(require("zod/v4-mini"));
const primitives_js_1 = require("../../lib/primitives.js");
const schemas_js_1 = require("../../lib/schemas.js");
/** @internal */
exports.OrganizationCustomerEmailSettings$inboundSchema = z.pipe(z.object({
    order_confirmation: z.boolean(),
    subscription_cancellation: z.boolean(),
    subscription_confirmation: z.boolean(),
    subscription_cycled: z.boolean(),
    subscription_past_due: z.boolean(),
    subscription_revoked: z.boolean(),
    subscription_uncanceled: z.boolean(),
    subscription_updated: z.boolean(),
}), z.transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "order_confirmation": "orderConfirmation",
        "subscription_cancellation": "subscriptionCancellation",
        "subscription_confirmation": "subscriptionConfirmation",
        "subscription_cycled": "subscriptionCycled",
        "subscription_past_due": "subscriptionPastDue",
        "subscription_revoked": "subscriptionRevoked",
        "subscription_uncanceled": "subscriptionUncanceled",
        "subscription_updated": "subscriptionUpdated",
    });
}));
/** @internal */
exports.OrganizationCustomerEmailSettings$outboundSchema = z.pipe(z.object({
    orderConfirmation: z.boolean(),
    subscriptionCancellation: z.boolean(),
    subscriptionConfirmation: z.boolean(),
    subscriptionCycled: z.boolean(),
    subscriptionPastDue: z.boolean(),
    subscriptionRevoked: z.boolean(),
    subscriptionUncanceled: z.boolean(),
    subscriptionUpdated: z.boolean(),
}), z.transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        orderConfirmation: "order_confirmation",
        subscriptionCancellation: "subscription_cancellation",
        subscriptionConfirmation: "subscription_confirmation",
        subscriptionCycled: "subscription_cycled",
        subscriptionPastDue: "subscription_past_due",
        subscriptionRevoked: "subscription_revoked",
        subscriptionUncanceled: "subscription_uncanceled",
        subscriptionUpdated: "subscription_updated",
    });
}));
function organizationCustomerEmailSettingsToJSON(organizationCustomerEmailSettings) {
    return JSON.stringify(exports.OrganizationCustomerEmailSettings$outboundSchema.parse(organizationCustomerEmailSettings));
}
function organizationCustomerEmailSettingsFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.OrganizationCustomerEmailSettings$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'OrganizationCustomerEmailSettings' from JSON`);
}
