// @polar-sh/sdk/dist/esm/models/components/customerwithmembers.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod/v4-mini";
import { remap as remap$ } from "../../lib/primitives.js";
import { safeParse } from "../../lib/schemas.js";
import { smartUnion } from "../../types/smartUnion.js";
import { Address$inboundSchema } from "./address.js";
import { CustomerType$inboundSchema } from "./customertype.js";
import { Member$inboundSchema } from "./member.js";
import { MetadataOutputType$inboundSchema, } from "./metadataoutputtype.js";
import { TaxIDFormat$inboundSchema } from "./taxidformat.js";
/** @internal */
export const CustomerWithMembersTaxId$inboundSchema = smartUnion([z.string(), TaxIDFormat$inboundSchema]);
export function customerWithMembersTaxIdFromJSON(jsonString) {
    return safeParse(jsonString, (x) => CustomerWithMembersTaxId$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CustomerWithMembersTaxId' from JSON`);
}
/** @internal */
export const CustomerWithMembers$inboundSchema = z.pipe(z.object({
    id: z.string(),
    created_at: z.pipe(z.iso.datetime({ offset: true }), z.transform(v => new Date(v))),
    modified_at: z.nullable(z.pipe(z.iso.datetime({ offset: true }), z.transform(v => new Date(v)))),
    metadata: z.record(z.string(), MetadataOutputType$inboundSchema),
    external_id: z.nullable(z.string()),
    email: z.string(),
    email_verified: z.boolean(),
    type: z.optional(z.nullable(CustomerType$inboundSchema)),
    name: z.nullable(z.string()),
    billing_address: z.nullable(Address$inboundSchema),
    tax_id: z.nullable(z.array(z.nullable(smartUnion([z.string(), TaxIDFormat$inboundSchema])))),
    locale: z.optional(z.nullable(z.string())),
    organization_id: z.string(),
    deleted_at: z.nullable(z.pipe(z.iso.datetime({ offset: true }), z.transform(v => new Date(v)))),
    members: z.optional(z.array(Member$inboundSchema)),
    avatar_url: z.string(),
}), z.transform((v) => {
    return remap$(v, {
        "created_at": "createdAt",
        "modified_at": "modifiedAt",
        "external_id": "externalId",
        "email_verified": "emailVerified",
        "billing_address": "billingAddress",
        "tax_id": "taxId",
        "organization_id": "organizationId",
        "deleted_at": "deletedAt",
        "avatar_url": "avatarUrl",
    });
}));
export function customerWithMembersFromJSON(jsonString) {
    return safeParse(jsonString, (x) => CustomerWithMembers$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CustomerWithMembers' from JSON`);
}
