// @polar-sh/sdk/dist/esm/models/components/customer.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod/v4-mini";
import { remap as remap$ } from "../../lib/primitives.js";
import { safeParse } from "../../lib/schemas.js";
import { smartUnion } from "../../types/smartUnion.js";
import { Address$inboundSchema, Address$outboundSchema, } from "./address.js";
import { CustomerType$inboundSchema, CustomerType$outboundSchema, } from "./customertype.js";
import { MetadataOutputType$inboundSchema, MetadataOutputType$outboundSchema, } from "./metadataoutputtype.js";
import { TaxIDFormat$inboundSchema, TaxIDFormat$outboundSchema, } from "./taxidformat.js";
/** @internal */
export const CustomerTaxId$inboundSchema = smartUnion([z.string(), TaxIDFormat$inboundSchema]);
/** @internal */
export const CustomerTaxId$outboundSchema = smartUnion([z.string(), TaxIDFormat$outboundSchema]);
export function customerTaxIdToJSON(customerTaxId) {
    return JSON.stringify(CustomerTaxId$outboundSchema.parse(customerTaxId));
}
export function customerTaxIdFromJSON(jsonString) {
    return safeParse(jsonString, (x) => CustomerTaxId$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CustomerTaxId' from JSON`);
}
/** @internal */
export const Customer$inboundSchema = z.pipe(z.object({
    id: z.string(),
    created_at: z.pipe(z.iso.datetime({ offset: true }), z.transform(v => new Date(v))),
    modified_at: z.nullable(z.pipe(z.iso.datetime({ offset: true }), z.transform(v => new Date(v)))),
    metadata: z.record(z.string(), MetadataOutputType$inboundSchema),
    external_id: z.nullable(z.string()),
    email: z.string(),
    email_verified: z.boolean(),
    type: z.optional(z.nullable(CustomerType$inboundSchema)),
    name: z.nullable(z.string()),
    billing_address: z.nullable(Address$inboundSchema),
    tax_id: z.nullable(z.array(z.nullable(smartUnion([z.string(), TaxIDFormat$inboundSchema])))),
    locale: z.optional(z.nullable(z.string())),
    organization_id: z.string(),
    deleted_at: z.nullable(z.pipe(z.iso.datetime({ offset: true }), z.transform(v => new Date(v)))),
    avatar_url: z.string(),
}), z.transform((v) => {
    return remap$(v, {
        "created_at": "createdAt",
        "modified_at": "modifiedAt",
        "external_id": "externalId",
        "email_verified": "emailVerified",
        "billing_address": "billingAddress",
        "tax_id": "taxId",
        "organization_id": "organizationId",
        "deleted_at": "deletedAt",
        "avatar_url": "avatarUrl",
    });
}));
/** @internal */
export const Customer$outboundSchema = z.pipe(z.object({
    id: z.string(),
    createdAt: z.pipe(z.date(), z.transform(v => v.toISOString())),
    modifiedAt: z.nullable(z.pipe(z.date(), z.transform(v => v.toISOString()))),
    metadata: z.record(z.string(), MetadataOutputType$outboundSchema),
    externalId: z.nullable(z.string()),
    email: z.string(),
    emailVerified: z.boolean(),
    type: z.optional(z.nullable(CustomerType$outboundSchema)),
    name: z.nullable(z.string()),
    billingAddress: z.nullable(Address$outboundSchema),
    taxId: z.nullable(z.array(z.nullable(smartUnion([z.string(), TaxIDFormat$outboundSchema])))),
    locale: z.optional(z.nullable(z.string())),
    organizationId: z.string(),
    deletedAt: z.nullable(z.pipe(z.date(), z.transform(v => v.toISOString()))),
    avatarUrl: z.string(),
}), z.transform((v) => {
    return remap$(v, {
        createdAt: "created_at",
        modifiedAt: "modified_at",
        externalId: "external_id",
        emailVerified: "email_verified",
        billingAddress: "billing_address",
        taxId: "tax_id",
        organizationId: "organization_id",
        deletedAt: "deleted_at",
        avatarUrl: "avatar_url",
    });
}));
export function customerToJSON(customer) {
    return JSON.stringify(Customer$outboundSchema.parse(customer));
}
export function customerFromJSON(jsonString) {
    return safeParse(jsonString, (x) => Customer$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'Customer' from JSON`);
}
