// @polar-sh/sdk/dist/esm/models/components/membersession.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod/v4-mini";
import { remap as remap$ } from "../../lib/primitives.js";
import { safeParse } from "../../lib/schemas.js";
import { Customer$inboundSchema } from "./customer.js";
import { Member$inboundSchema } from "./member.js";
/** @internal */
export const MemberSession$inboundSchema = z.pipe(z.object({
    created_at: z.pipe(z.iso.datetime({ offset: true }), z.transform(v => new Date(v))),
    modified_at: z.nullable(z.pipe(z.iso.datetime({ offset: true }), z.transform(v => new Date(v)))),
    id: z.string(),
    token: z.string(),
    expires_at: z.pipe(z.iso.datetime({ offset: true }), z.transform(v => new Date(v))),
    return_url: z.nullable(z.string()),
    member_portal_url: z.string(),
    member_id: z.string(),
    member: Member$inboundSchema,
    customer_id: z.string(),
    customer: Customer$inboundSchema,
}), z.transform((v) => {
    return remap$(v, {
        "created_at": "createdAt",
        "modified_at": "modifiedAt",
        "expires_at": "expiresAt",
        "return_url": "returnUrl",
        "member_portal_url": "memberPortalUrl",
        "member_id": "memberId",
        "customer_id": "customerId",
    });
}));
export function memberSessionFromJSON(jsonString) {
    return safeParse(jsonString, (x) => MemberSession$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'MemberSession' from JSON`);
}
